arquillian-chameleon-runner/runner/src/main/java/org/arquillian/container/chameleon/runner/SystemPropertyResolver.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2016 pact-jvm project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.arquillian.container.chameleon.runner;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class SystemPropertyResolver {
```

Comment:
Quite a bit of code w/o a unit test. I think we also have this kind of logic in `-core`. Maybe worth checking instead of borrowing from Pact?

Reply:
I thought the same but there is one problem and it is that the class that does this logic is packaged scope, so cannot be used outside. The same logic is used in algeron, cube and chameleon, so what I would do is accept this PR with this class, and open a new issue in organization repo so when this PR is merged https://github.com/arquillian/arquillian-core/pull/162 then use this logic in all these issues.

Reply:
Ok, let's do that this way